### ISSUE TITLE
Get run time error in Angular app with ng 4 and Cli

### ISSUE THREAD

**Issue by mohsenvafa** [oktadeveloper/okta-angular-sign-in-widget-example#5]
Using @ okta/okta-signin-widget version 2.1.0 in an Angular 4 app created by Cli gets following run time error when it is built the app in prod mode (ng build --prod -env=prod)
ERROR ReferenceError: _global is not defined

![image](https://user-images.githubusercontent.com/5973501/28593821-aa7279c2-7142-11e7-87d6-f393d4df8113.png)

**Comment by mraible** [comment]
When I try to run this command on the existing repository, I get the following error:

```
ERROR in ./src/main.ts
Module not found: Error: Can't resolve './$$_gendir/app/app.module.ngfactory' in '/Users/mraible/dev/okta/okta-angular-sign-in-widget-example/src'
 @ ./src/main.ts 4:0-74
 @ multi ./src/main.ts
```

**Comment by mohsenvafa** [comment]
@mraible do you mean (ng build --prod) command?

**Comment by mraible** [comment]
I get the same error after upgrading to `@okta/okta-signin-widget` version 2.1.0 and running `ng build --prod`. My package.json changes:

```diff
diff --git a/package.json b/package.json
index 1297fe6..f737d6f 100644
--- a/package.json
+++ b/package.json
@@ -20,7 +20,7 @@
     "@angular/platform-browser": "^4.0.0",
     "@angular/platform-browser-dynamic": "^4.0.0",
     "@angular/router": "^4.0.0",
-    "@okta/okta-signin-widget": "^1.9.0",
+    "@okta/okta-signin-widget": "2.1.0",
     "core-js": "^2.4.1",
     "rxjs": "^5.1.0",
     "zone.js": "^0.8.4"
```

**Comment by mraible** [comment]
@mohsenvafa I created a new app from scratch using the latest version of Angular CLI and the widget. I'm unable to reproduce this issue. You can see my app at the following URL:

https://github.com/mraible/tmp-angular-signin-widget-example

Here's the output I get:

```
± ng build --prod -env=prod
 23% building modules 110/112 modules 2 active ...ignin-widget-example/src/polyfills.tsThe final argument to magicString.overwrite(...) should be an options object. See https://github.com/rich-harris/magic-string
Hash: c08a1f8af45077598ecc
Time: 12193ms
chunk    {0} polyfills.71b130084c52939ca448.bundle.js (polyfills) 177 kB {5} [initial] [rendered]
chunk    {1} main.35350d4d37c571e32632.bundle.js (main) 17.5 kB {4} [initial] [rendered]
chunk    {2} scripts.cf6129aaeb7b9e490517.bundle.js (scripts) 1.02 MB {5} [initial] [rendered]
chunk    {3} styles.399923ac4fe08cddb827.bundle.css (styles) 175 bytes {5} [initial] [rendered]
chunk    {4} vendor.275f652721229c45fbe3.bundle.js (vendor) 849 kB [initial] [rendered]
chunk    {5} inline.d69aacd28e29b47d2df5.bundle.js (inline) 0 bytes [entry] [rendered]
```

**Comment by mohsenvafa** [comment]
@mraible I don't get this error in build time, I get this error in run time after build and deploy to IIS. When I navigate to Login page, this error is shown.
I am using angular 4.3.1

**Comment by mraible** [comment]
@mohsenvafa Upgrading to the latest dependencies in `package.json` seems to solve the problem. I created PR #6 to show what needs to change. 

Steps I performed to confirm this works:

1. Cloned this project, checked out the branch for the PR: `git checkout signin-widget-2.1.0`
2. Ran `npm install`
3. Ran `ng build --prod -env=prod`
4. cd into `dist` and ran `python -m SimpleHTTPServer 4200`

I was able to log in at <http://localhost:4200> and didn't see any errors in my console. When I try this on the master branch of this project, I get the following error when running `ng build --prod -env=prod`.

```
ERROR in ./src/main.ts
Module not found: Error: Can't resolve './$$_gendir/app/app.module.ngfactory' in '/Users/mraible/dev/okta/okta-angular-sign-in-widget-example/src'
 @ ./src/main.ts 4:0-74
 @ multi ./src/main.ts
```
